File: kun-metadata/src/main/java/com/miotech/kun/metadata/constant/DataStoreType.java
package com.miotech.kun.metadata.constant;

public enum DataStoreType {
    TABLE("PostgresTable/MySQLTable/HiveTable"),
    INDEX("ElasticsearchIndex"),
    COLLECTION("ArangoCollection/MongoCollection"),
    TOPIC("KafkaTopic"),
    SHEET("GoogleSpreadSheet"),
    FILE("File");

    private String type;

    DataStoreType(String type) {
        this.type = type;
    }
}

File: kun-frontend/src/definitions/DataStore.type.ts
export type DataStoreTypeEnum =
  | 'ARANGO_COLLECTION'
  | 'ELASTICSEARCH_INDEX'
  | 'FILE'
  | 'HIVE_TABLE'
  | 'MONGO_COLLECTION'
  | 'MYSQL_TABLE'
  | 'POSTGRES_TABLE'
  | 'SHEET'
  | 'TOPIC';

export interface DataStore {
  type: DataStoreTypeEnum;
}
